File: Array/643-Maximum-Average-Subarray-I.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int maxAverage = 0;

        for (int i = 0; i < k; i++) {
            maxAverage += nums[i];
        }

        int sum = maxAverage;
        System.out.println("This is the sum: " + sum);

        for (int i = k; i < nums.length; i++) {
            sum = sum + nums[i] - nums[i - k];
            maxAverage = Math.max(maxAverage, sum);
        }

        return (double) maxAverage / k;
    }
}

File: Array/643-Maximum-Average-Subarray-I.ts
function findMaxAverage(nums: number[], k: number): number {
    let sumArray: number = 0;

    for (let i = 0; i < nums.length; i++) {
        sumArray += nums[i];
    }

    let maxValue = 0;

    for (let i = k; i < nums.length; i++) {
        sumArray = sumArray + nums[i] - nums[i - k];
        maxValue = Math.max(maxValue, sumArray);
    }

    return maxValue / k;
};